Mutant dreams

Jerimee opened the pull request:
#### Summary

SUMMARY: Content "more dreams for mutants"

#### Purpose of change

Add more dream content for mutants via JSON, especially dreams for post-threshold mouse.  I got sick of seeing the same two dreams over and over and over, so I scratched my own itch and wrote some new ones.

#### Describe the solution

A handful of new dreams for level 4 Chimera, a dozen more for level 4 Mouse, and a handful for few other branches.

Trog now has a dream (and I removed the bit about the knockers): _You dream you crawl out from under a rock.  And then immediately crawl back in._

I think all the cattle ones have been removed with the following exception: _You belllow on a knoll against the sky, having feasted on the apple time cider._ So, with regard to cattle, this PR adds just that one.

New level 4 mouse dreams:

1. "Wolf, hawk, fox, and snake can't stand in my way.  My body is weak and it may break, but not today.",
2. "Oh boy! This hidey hole sure is swell.",
3. "You dream of tiny crowds cheering, \"Chew, gnaw, scratch and chew.  Slow and sure 'til we break through!"\",
4. "The heist of the century!  Once we break into that grain bunker we'll be on easy street.",
5. "Once we finish the burrow we'll settle down and start a little family.  Maybe just a dozen boys and a dozen girls.  Nice and cosy.",
6. "As you sleep you dream of… swiss cheese ice cream.  Feed pellets and kernels of corn, with moldy bread.  Oh boy, what a feast!",
7. "In your dreams, you blindly run through the pipes of the city, navigating the familiar byways via smell alone.",
8.  "In your sleep your mother squeaks to you, \"Beware the wisdom of the owl and beware the vision of the hawk.\"",
9.  "Sneaky sneak under the floorboards.  We will sneak all their crumbs for us.  But later, back to sleep for now.",

#### Describe alternatives you've considered

The dream content is a bit lopsided now, level 4 mouse has more, where as many branches are unchanged.

I did not modify the existing dreams, some of which do not meet the expectations expressed in the comments on this PR.

~~Via wapcatlet: _I have a slight concern that, taken out of context, it may not be clear these messages are about dreams. When the player gets up for a snack while their character sleeps, and comes back to a message like "Your first arrow hits the bullseye" or "Hot dog! You find ANOTHER grain of rice", confusion may ensue._  (so what I need to do is look through the new ones and, for any that are confusing, add something explicit about dreaming or sleeping)~~

~~Feedback I got from a contributor: _More dreams is actually kind of a pain in the ass as they provide necessary gameplay information to the player and burying them under a dozen strings of flavor text makes it more of a hassle without adding much_  When I started this PR I was mostly just thinking about level 4 dreams, which are simply flavor text.  I agree that before level 4 the dreams serve a game function as a clue about how one is mutating and that clue shouldn't be esoteric (ie it needs to be a legit clue)~~

#### Additional context

The following Reddit users contributed content (big thanks!): 

- u/CCC_037 
- u/Sytruan
- u/Stretop

https://www.reddit.com/r/cataclysmdda/comments/ie9ut9/mutant_dreams/

Thanks to Rivet, Fris0uman, Candlebury, KG, wapcaplet, Molly, and others for the edits, feedback, and reviews.

Rivet-the-Zombie commented on data/json/dreams.json:
```diff
@@ -106,7 +106,15 @@
+      "You have a strange dream.",
+      "You have a puckish dream.",
+      "You have a playful dream.",
+      "You have a strange dream about orcs.",
```
ELF-A mutants may be sort of tree-people, but they're definitely not Tolkien elves, and don't have anything to do with orcs.

Rivet-the-Zombie commented on data/json/dreams.json:
```diff
@@ -276,7 +290,14 @@
-    "messages": [ "You dream of a home in the forests.", "You feel beautiful, and yet riven with worry..." ],
+    "messages": [
+      "You dream of a home in the forests.",
+      "You dream of drinking goblin wine.",
```
...or goblins for that matter.

Rivet-the-Zombie commented on data/json/dreams.json:
```diff
@@ -449,20 +471,30 @@
+    "messages": [
+      "NO!  You will not allow this corruption to prevail!",
+      "You see yourself reflected in the beauty of the forest.",
+      "While fearful, you dare not turn down an invitation to visit the Unseelie Court."
```
They're mutants that are vaguely similar to literary elves. They're not actually mythological creatures.

Jerimee replied:
Yeah, I was thinking that an evolving mutant would still have dreams influenced by their human cultural legacy.  Joseph Campbell or Jungian archetypes

_"innate universal pre-conscious psychic dispositions that form the substrate from which the basic themes of human life emerge"_

Rivet-the-Zombie commented on data/json/dreams.json:
```diff
@@ -545,6 +579,8 @@
     "messages": [
       "You dream of the hunt, the feed, and your pack.",
+      "You lead the pack on a glorious hunt, knowing that they will all have your back as you have theirs!",
+      "You ride a black horse.  It is the end of days.  You are a salesman; one of a pack of four.  You sell only hunger, and you go to every door.",
```
What does the Abrahamic eschaton scenario have to do with mutating into a partial-wolf-thing?

Jerimee replied:
"The wolf at the door" = famine

But I'll remove it

Rivet-the-Zombie commented:
I feel like a lot of the mouse ones are particularly on the nose, and seem more like the player is channeling the specific memories of a dead fancy rat than dreaming about what it would be like to actually live as such a creature.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -206,6 +214,7 @@
     "type": "dream",
     "messages": [
       "You dream of grazing in an open field.",
+      "You dream of a golden calf.",
```
It goes straight from dreaming of being a cow to mystic dream about false idoles? I don't think this line fits the tone ^^

Jerimee replied:
ok, removing the golden calf ones and references to labrynth

Fris0uman commented on data/json/dreams.json:
```diff
@@ -224,6 +233,7 @@
     "type": "dream",
     "messages": [
       "You have a confusing dream of growing in a garden.",
+      "You remember a strange song lyric you once heard.  \"The ardor of arboreality is an adventure we have spurned\" - it is clearly time to overturn a new leaf.",
```
This might be overcomplicated. Most of the mutant dreams are about being the thing you're turning into, so I think plant dreams for exemple could be about craving sunlight, feeling the wind in your leafs, fearing the bugs eating you, spreading your roots in the soill etc... but not really about cryptic messages from song lyrics.

Jerimee replied:
makes sense

Fris0uman commented on data/json/dreams.json:
```diff
@@ -237,7 +247,11 @@
-    "messages": [ "You dream of living deep inside a dark cave.", "You dream of being a primitive cave dweller." ],
+    "messages": [
+      "You dream of living deep inside a dark cave.",
+      "The Tommyknockers are our friends.  They warn us of danger.  We'll leave a little cake for them on this rock.",
```
Google tells me this is a stephen king reference, so deos this assume that every character has read the novel? More importantly players that haven't read it (like me) will have no idea what this mean. Again the dreams we have currently are about being the thing you're becoming and I think sticking to that would be good.

Jerimee replied:
while King did write a book, tommyknockers are not his creation - tommyknockers are the ghosts of dead miners who warn those traveling under ground of danger (ie, mine collapse, pitfall) - https://www.legendsofamerica.com/gh-tommyknockers/

Fris0uman replied:
Is it a common thing in new england? Still this feels  out of place in the mutant dreams.

Jerimee replied:
ok, I will remove it

Fris0uman commented on data/json/dreams.json:
```diff
@@ -376,6 +397,7 @@
   {
     "type": "dream",
     "messages": [
+      "You have a nightmare of twisting, endless corridors⋯  They couldn't kill you so they built a labyrinth to cage you.",
```
Again you're becoming more like cattle but not an actual minotaur.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -554,6 +590,9 @@
     "messages": [
       "You dream of jumping the gate and leading your herd to freedom.",
+      "You belllow  on a knoll against the sky, having feasted on the apple time cider.",
+      "Yes, let them worship the golden calf.  Let them worship their bovine masters.",
```
Same here golden calf and minotaure references feel out of place.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -563,6 +602,8 @@
     "type": "dream",
     "messages": [
       "You feel smug, knowing that the Queen is dependent on you.",
+      "You relax and enjoy the humming adulation of all your followers.",
```
The dream above is about being a worker and this one about being the queen but they're both in the same category?

Jerimee replied:
Yeah, but it's a dream, it doesn't have to be logical

besides, a person evolving into an entity with insect like features isn't necessarily going to have exact specific behavior of a given insect species

Jerimee replied:
Fris0uman wrote _**"Nitpick but I'm pretty sure no eusocial insect does that, the queen does not really gives order and wouldn't know if an intruder came into the colony."**_

I just wanted to move that comment here, so it can resolved with the other insect dreams.

Fris0uman replied:
> Yeah, but it's a dream, it doesn't have to be logical

OK, fair enough.

My other comment was just that a queen bee or an ant queen would not send guards gettting rid of an intruder, it barely ever interacts with guards and it would have no idea if an intruder came in the hive because it lives far away from the entrance.  But you can chose to ignore that, maybe mutation dream don't have to be 100% accurate about animal behaviour ^^"

Fris0uman commented on data/json/dreams.json:
```diff
@@ -563,6 +602,8 @@
     "messages": [
       "You feel smug, knowing that the Queen is dependent on you.",
+      "You relax and enjoy the humming adulation of all your followers.",
+      "Something dangerous has entered your area!  You send a couple of hundred followers to deal with it.",
```
Nitpick but I'm pretty sure no eusocial insect does that, the queen does not really gives order and wouldn't know if an intruder came into the colony.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -639,14 +683,24 @@
+      "Your body flows slightly faster than you expected… oh, just a dream.",
+      "In your dream you look into the abyss between dimensions. It looks back… and is very confused.",
+      "You gaze long into the abyss.  The abyss whispers \"Now I am become monsterous.\"  You are talking to yourself again.",
+      "Your predator parts dream about chasing down your less-predator parts.",
```
Again those feel a tad too mystical.

Jerimee replied:
What good is a Chimera if it can't be mystical?  I actually didn't write these, but I still like them.  The reference is from Nietzsche (sp?): _"Whoever fights with monsters should see to it that he does not become a monster in the process. And when you gaze long into an abyss the abyss also gazes into you."_

Fris0uman replied:
Maybe the abyss thing is fine, it does convey some of the confusion that seems to be associated with the chimera line.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -660,6 +714,25 @@
+      "Wolf, hawk, fox, and snake can't stand in my way.  My body is weak and it may break, but not today.",
+      "Oh boy! This hidey hole sure is swell.",
+      "Run, run, run, run, run, run, stop!  Smell⋯  Listen⋯  Run, run, run, run, stop!  Smell that?  Food!  Run, run, run, run.",
+      "Chew, gnaw, bite and chew.  We'll make quick work of these ropes and let loose the lion!",
```
> We'll make quick work of these ropes and let loose the lion!",

wat?

Jerimee replied:
Aesop's fable demonstrating the power of the mouse - https://fablesofaesop.com/the-lion-and-the-mouse.html

Jerimee replied:
Did you still want to remove the aesop fable one about chewing through the ropes?

Fris0uman reviewed (commented):
So overall I think you've made a lot of those too complicated and mystic.  

>     "You have a nightmare about being washed out of a drainpipe.",
>     "You relax in your web; it's not vibrating, so dinner hasn't arrived yet.",

Those two for exemple are excellent and I think that's more the tone mutant dreams try to have.

Fris0uman commented on data/json/dreams.json:
```diff
@@ -376,6 +391,7 @@
   {
     "type": "dream",
     "messages": [
+      "You have a nightmare of twisting, endless corridors⋯",
```
Still feels pretty far from cattle

Jerimee replied:
Yeah I didn't write this one (it came from user on reddit) so I wanted to preserve it if possible.  I figured this way it could be either a minotaur or a slaughterhouse stockade?

Ok for me to chat you on discord to discuss the remaining open conversations? Most of them I've closed simply by deleting the offending content, or incorporating feedback, but Chimera, Insect, and Mouse are still open.

kevingranade replied:
TBF all of the cattle dreams are problematic.  It needs to be about things cattle DO, not things that are done to cattle.

John-Candlebury commented on data/json/dreams.json:
```diff
@@ -536,6 +563,7 @@
     "type": "dream",
     "messages": [
       "Boing!  and you've pounced another yummy critter.",
+      "You purr as your slave scratches you between the ears.",
```
```suggestion
      "You purr as your pet scratches you between the ears.",
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -276,7 +286,12 @@
-    "messages": [ "You dream of a home in the forests.", "You feel beautiful, and yet riven with worry..." ],
+    "messages": [
+      "You dream of a home in the forests.",
+      "You dream of drinking flowered wine.",
```
This is crossing the line into fantasy elf.
```suggestion
```

Jerimee replied:
Yeah I think there is some canon I'm missing here - like what type of fairie folk are not fantasy?  These Elf-A aren't really elves, they are plant bird people?

kevingranade commented on data/json/dreams.json:
```diff
@@ -503,6 +529,7 @@
     "type": "dream",
     "messages": [
       "You wonder if your nest will be good enough.",
+      "You swoop down from the skies, and somehow manage to lift off carrying an entire sheep!",
```
This is explicitly something that can not happen in the game, let's not be intentionally misleading.
```suggestion
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -536,6 +563,7 @@
     "type": "dream",
     "messages": [
       "Boing!  and you've pounced another yummy critter.",
+      "You purr as your pet scratches you between the ears.",
```
Again this is something done to cats, not something cats do.
```suggestion
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -449,20 +463,30 @@
     "type": "dream",
     "messages": [
       "You see yourself, five years from now, as the leader of a successful city.",
+      "Obeying orders, you revel in the satisfaction of a job well done.",
```
I don't see how this relates to alpha at all.

kevingranade commented on data/json/dreams.json:
```diff
@@ -612,6 +640,8 @@
     "type": "dream",
     "messages": [
       "You excitedly web up an interloper and prepare to feast… nope, dream.",
+      "You have a nightmare about being washed out of a drainpipe.",
```
Nursery rhyme references are a bit much.
```suggestion
```

Leland replied:
Aw, I liked this one

kevingranade commented on data/json/dreams.json:
```diff
@@ -639,14 +670,24 @@
     "type": "dream",
     "messages": [
       "You pant, terrified at the thought of that FUNGUS destroying your home!",
+      "Your first arrow hits the bullseye.  Your second arrow splits the first arrow in half.",
+      "You dance easily across a tightrope, never in any danger of falling.",
+      "You encourage your followers to be independent.  The more new ideas you try, the more chances that you'll find the best ideas!",
```
These are creeping a bit too far into the fantasy elf trope.

Jerimee replied:
deleting these then

wapcaplet commented:
@Jerimee These are great; I have always found messages like this to be a big part of my enjoyment of CDDA, so as a player I very much appreciate the work you are doing to add linguistic flavor. Thanks!

I have a slight concern that, taken out of context, it may not be clear these messages are about dreams. When the player gets up for a snack while their character sleeps, and comes back to a message like "Your first arrow hits the bullseye" or "Hot dog!  You find ANOTHER grain of rice", confusion may ensue.

Most of the messages are poetic enough to be clearly unreal, but I'd be careful of any that could be taken too literally.

Jerimee commented:
In addition to the previous changes, I've now gone through and changed any new 1-3 dreams to be more straightforward and indicative of their mutation paths.

Additionally I've deleted or modified half of the level 4 mouse dreams to make them more clearly dreams and remove cultural allusions. 

There is now significantly less new content here, but the content is much improved.

Leland commented:
> I'd be careful of any that could be taken too literally

We could add in a log entry that says something to the order of "Looks like it was just a dream..." after sleep that features dreams

kevingranade commented on data/json/dreams.json:
```diff
@@ -449,20 +462,30 @@
     "type": "dream",
     "messages": [
       "You see yourself, five years from now, as the leader of a successful city.",
+      "Obeying orders, you revel in the satisfaction of a job well done.",
```
```suggestion
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -563,6 +587,8 @@
     "messages": [
       "You feel smug, knowing that the Queen is dependent on you.",
+      "You relax and enjoy the humming adulation of all your followers.",
+      "Something dangerous has entered your area!  You send a couple of hundred followers to deal with it.",
```
```suggestion
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -646,7 +673,14 @@
   {
     "type": "dream",
-    "messages": [ "Your body flows slightly faster than you expected… oh, just a dream.", "FIGHT.  FEED.  FORWARD." ],
+    "messages": [
+      "Your body flows slightly faster than you expected… oh, just a dream.",
+      "In your dream you look into the abyss between dimensions. It looks back… and is very confused.",
+      "You gaze long into the abyss.  The abyss whispers, \"Now I am become monstrous.\"  You are talking to yourself again.",
+      "Your predator parts dream about chasing down your less-predator parts.",
+      "Three headed and snarling, you guard hell from the horrors of the living world.",
+      "FIGHT.  FEED.  FORWARD."
+    ],
```
```suggestion
     "messages": [ "Your body flows slightly faster than you expected… oh, just a dream.", "FIGHT.  FEED.  FORWARD." ],
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -646,7 +673,7 @@
   {
     "type": "dream",
-    "messages": [ "Your body flows slightly faster than you expected… oh, just a dream.", "FIGHT.  FEED.  FORWARD." ],
+     "messages": [ "Your body flows slightly faster than you expected… oh, just a dream.", "FIGHT.  FEED.  FORWARD." ],
```
```suggestion
    "messages": [ "Your body flows slightly faster than you expected… oh, just a dream.", "FIGHT.  FEED.  FORWARD." ],
```

kevingranade commented on data/json/dreams.json:
```diff
@@ -660,6 +687,15 @@
     "type": "dream",
     "messages": [
       "You curl up some more in your nest.  Geez, shredded paper really *is* comfy.",
+      "Wolf, hawk, fox, and snake can't stand in my way.  My body is weak and it may break, but not today.",
+      "Oh boy! This hidey hole sure is swell.",
+      "You dream of tiny crowds cheering, \"Chew, gnaw, scratch and chew.  Slow and sure 'til we break through!\"",
+      "The heist of the century!  Once we break into that grain bunker we'll be on easy street.",
+      "Once we finish the burrow we'll settle down and start a little family.  Maybe just a dozen boys and a dozen girls.  Nice and cosy.",
+      "As you sleep you dream of… swiss cheese ice cream.  Feed pellets and kernels of corn, with moldy bread.  Oh boy, what a feast!",
+      "In your dreams, you blindly run through the pipes of the city, navigating the familiar byways via smell alone.",
+      "In your sleep your mother squeaks to you, \"Beware the wisdom of the owl and beware the vision of the hawk.\"",
+      "Sneaky sneak under the floorboards.  We will sneak all their crumbs for us.  But later, back to sleep for now.",
```
```suggestion
      "Oh boy! This hidey hole sure is swell.",
      "Once we finish the burrow we'll settle down and start a little family.  Maybe just a dozen boys and a dozen girls.  Nice and cosy.",
      "In your dreams, you blindly run through the pipes of the city, navigating the familiar byways via smell alone.",
      "Sneaky sneak under the floorboards.  We will sneak all their crumbs for us.  But later, back to sleep for now.",
```
